core/search/sort.hpp

```diff
@@ -361,13 +406,150 @@ struct score_traits {
+                              score_cast(src_start[1] + offset));
+        break;
+      case 3:
+        casted_dst = std::max(
+                       std::max(score_cast(src_start[0] + offset),
+                                score_cast(src_start[1] + offset)),
+                       score_cast(src_start[2] + offset));
+        break;
+      case 4:
+        casted_dst = std::max(
+                       std::max(score_cast(src_start[0] + offset),
+                                score_cast(src_start[1] + offset)),
+                       std::max(score_cast(src_start[2] + offset),
+                                score_cast(src_start[3] + offset)));
+        break;
+      case 5:
+        casted_dst = std::max(
+                       std::max(std::max(score_cast(src_start[0] + offset),
+                                         score_cast(src_start[1] + offset)),
+                                std::max(score_cast(src_start[2] + offset),
+                                         score_cast(src_start[3] + offset))),
+                       score_cast(src_start[4] + offset));
+        break;
+      case 6:
+        casted_dst = std::max(
+                       std::max(std::max(score_cast(src_start[0] + offset),
+                                         score_cast(src_start[1] + offset)),
+                                std::max(score_cast(src_start[2] + offset),
+                                         score_cast(src_start[3] + offset))),
+                       std::max(score_cast(src_start[4] + offset),
+                                score_cast(src_start[5] + offset)));
+        break;
+      case 7:
+        casted_dst = std::max(
+                       std::max(std::max(score_cast(src_start[0] + offset),
+                                         score_cast(src_start[1] + offset)),
+                                std::max(score_cast(src_start[2] + offset),
+                                         score_cast(src_start[3] + offset))),
+                       std::max(std::max(score_cast(src_start[4] + offset),
+                                         score_cast(src_start[5] + offset)),
+                                score_cast(src_start[6] + offset)));
+
+        break;
+      case 8:
+        casted_dst = std::max(
+                       std::max(std::max(score_cast(src_start[0] + offset),
+                                         score_cast(src_start[1] + offset)),
+                                std::max(score_cast(src_start[2] + offset),
+                                         score_cast(src_start[3] + offset))),
+                       std::max(std::max(score_cast(src_start[4] + offset),
+                                         score_cast(src_start[5] + offset)),
+                                std::max(score_cast(src_start[6] + offset),
+                                         score_cast(src_start[7] + offset))));
+        break;
+      default:
+        casted_dst = score_cast(src_start[0] + offset);
```

Comment:
Is std::max_element() slower?

Reply:
I would need to write custom iterator to use `std::max_element` since I compare scores at a certain offset